dbcontrollers/products.go

```diff
@@ -169,13 +202,121 @@ func (*MYSQLController) GetProduct(productID *uuid.UUID) (*models.ProductData, e
 		Assets:  assets,
 	}
 
-	return &productData, nil
+	return &productData, mysqldb.DBConnector.Commit(tx)
+}
+
+func (*MYSQLController) UpdateProductDetails(productData *models.ProductData) error {
+
+	if err := mysqldb.UpdateAsset(mysqldb.ProductDetails, productData.Details); err != nil {
+		if fmt.Errorf(mysqldb.ErrAssetMissing, mysqldb.ProductDetails).Error() == err.Error() {
+			return ErrMissingProductDetail
+		}
+		return err
+	}
+	return nil
```

Comment:
It can make sense. But if 95% of the code has a success case as  the last return, wouldn't it be confusing that there is one function where success returns before the end?
I mean habits build up quickly and if there is an exception in the code layout developer may overlooks it more easily. I tend to try to code in a uniform manner.

Reply:
Depends on the viewpoint. If you always early-return in your methods and don't bother with the success or fail cases, you could decrease the complexity of your code.
So mainly, Complexity OVER Habits.